Spigot-Server-Patches/0722-Add-PlayerPickItemEvent.patch

```diff
@@ -0,0 +1,56 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Rodney_Mc_Kay <herr-fant@web.de>
+Date: Fri, 7 May 2021 21:46:28 +0200
+Subject: [PATCH] Add PlayerPickItemEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/network/PlayerConnection.java b/src/main/java/net/minecraft/server/network/PlayerConnection.java
+index 9455cb9bc849a330e57fdc466fb51902631e22d8..f209d2c23070e45c8d0158744062215f7b3776bc 100644
+--- a/src/main/java/net/minecraft/server/network/PlayerConnection.java
++++ b/src/main/java/net/minecraft/server/network/PlayerConnection.java
+@@ -904,7 +904,18 @@ public class PlayerConnection implements PacketListenerPlayIn {
+             this.disconnect("Invalid hotbar selection (Hacking?)");
+             return;
+         }
+-        this.player.inventory.c(packetplayinpickitem.b()); // Paper - Diff above if changed
++        // Paper end
++        // Paper start - PlayerPickItemEvent
++        Player bukkitPlayer = player.getBukkitEntity();
++        int targetSlot = this.player.inventory.getPickItemTargetSlot();
++        int sourceSlot = packetplayinpickitem.b();
++
++        io.papermc.paper.event.player.PlayerPickItemEvent event = new io.papermc.paper.event.player.PlayerPickItemEvent(bukkitPlayer, targetSlot, sourceSlot);
++        server.getPluginManager().callEvent(event);
++
++        if (event.isCancelled()) return;
```

Comment:
Can be combined into one line. The abstract `Event` class has a `callEvent` method that returns a boolean based on the cancellation state of the event.